spec.md

```diff
@@ -873,6 +883,51 @@ DELETE http://localhost:8080/csp/containers/v1/snapshot_groups/052265c9672660666
 204 No Content
 ```
 
+DELETE `/containers/v1/snapshot_groups?delete_downstream=true`
```

Comment:
Does this only delete the downstream snapshot_group?

Reply:
aah, missed to update the spec. This is the final decision we went with
```          
@Parameter(
                    description = "Authorization headers",
                    required = true)
            @ApiParam(
                    value = "Authorization headers",
                    hidden = true
            )
            @BeanParam RequestBean request,

            @Parameter(
                    description = "The id of the snapshot group to delete",
                    required = true)
            @PathParam("id")
                    String id,
            @Parameter(
                    description = "The name of a downstream replication partner from which to delete the snapshot group, with no effect upstream")
            @QueryParam("replication_partner")
                    String replicationPartner,
            @RequestBody(
                    description = "List of IDs of the snapshots in the downstream snapshot group",
                    content = @Content(
                            schema = @Schema(implementation = SnapshotGroup.class)))
                    List<String> snapshotIds)
```

Reply:
Why does the replication_partner need to be specified?  Can't we just look that up from the nimble array?

Reply:
We want to support deletion of remote snapshots independently and we need the list of snapshot ids and replication partner to achieve this. 
There is a possibility that the upstream snapshots could be deleted while only the downstream remains. Since the volumegroup (added to spec) and snapshotgroup already know about the replication partner and snapshot ids, we don't have to make additional calls

Reply:
Hmmm.  Why can't the API do the following:

```
SnapshotCollection snapColl = NimOS.getSnapshotCollection(id);
String partnerName = snapColl.getReplicateTo();
String partnerId = NimOS.getReplicationpartnerIdFromName(partnerName);
for (String snapshotId : sg.getSnapshotList()) {
    NimOS.deleteSnapshot(snapshotID, partnerID);
}
```

Reply:
```
snapColl = NimOS.getSnapshotCollection(id);
```
this would work if the upstream snapcoll still exists. We identified a scenario where its possible for upstream and downstream to be independent.

Reply:
Please elaborate on that scenario.  Upstream and downstream snapcolls are always connected AFAIK.  Otherwise, what's the purpose of supplying the id of the snapshot_group?  Why not just do "DELETE /containers/v1/snapshots/{id}?delete_downstream=true"?

Reply:
The lifecycle of the upstream/downstream snapcolls are independent after the replication is complete. 
We can still have downstream snapshots even though the upstream has been deleted (we would want to support this use case from our design discussions).
Now, to be able to delete downstream snapcolls from an upstream snapshots we can still achieve it if we feed it with enough information for replication_partner to invoke the delete on the downstream.
I have covered this in the Nimble CSP SnapshotGroupTest scenario as followis
```
        1. delete the local snapshotGroup
        2. the upstream snapshotgroup should have been deleted
        3. now delete the remote snapshotGroup 
        4. the delete should go through with the partner_name and list of snapshots.
```